core/src/main/scala/org/broadinstitute/workbench/hamm/pricing/JsonCodec.scala

```diff
@@ -1,12 +1,63 @@
 package org.broadinstitute.workbench.hamm
 package pricing
 
-import io.circe.Decoder
+import io.circe.{Decoder, DecodingFailure, Json}
 
 //TODO: this needs to be updated to use https://cloud.google.com/billing/v1/how-tos/catalog-api
 object JsonCodec {
-  implicit val cpuDecoder: Decoder[CpuCost] = Decoder.forProduct1("us")(CpuCost.apply)
-  implicit val ramDecoder: Decoder[RamCost] = Decoder.forProduct1("us")(RamCost.apply)
-  implicit val computeCostDecoder: Decoder[ComputeCost] = Decoder.forProduct2("CP-DB-PG-CUSTOM-VM-CORE", "CP-DB-PG-CUSTOM-VM-RAM")(ComputeCost.apply)
-  implicit val gcpPriceListDecoder: Decoder[GcpPriceList] = Decoder.forProduct1("gcp_price_list")(GcpPriceList.apply)
-}
\ No newline at end of file
+
+  implicit val categoryDecoder: Decoder[Category] = Decoder.instance { cursor =>
+    for {
+      resourceFamily <- cursor.downField("resourceFamily").as[ResourceFamily]
+      serviceDisplayName <- cursor.downField("serviceDisplayName").as[String]
+      resourceGroup <- cursor.downField("resourceGroup").as[String]
+      usageType <- cursor.downField("usageType").as[String]
+    } yield Category(ServiceDisplayName(serviceDisplayName), resourceFamily, ResourceGroup(resourceGroup), UsageType.stringToUsageType(usageType))
+  }
+
+  implicit val tieredRateDecoder: Decoder[TieredRate] = Decoder.instance { cursor =>
+    for {
+      startUsageAmount <- cursor.downField("startUsageAmount").as[Int]
```

Comment:
I would write decoder for all custom types instead of decoding them to primitive types and then wrap them in custom types

Reply:
That sounds like a lot more code than what here now - why would separate decoders be better when they aren't used anywhere else. Maybe it would be better if we wrote them separately later if there was a need?